daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/DSELConverter.java

```diff
@@ -8,28 +8,34 @@
 import org.talend.tql.model.Expression;
 import org.talend.tql.parser.Tql;
 
-public class DSELConverter {
+import java.util.HashMap;
+import java.util.Map;
 
-    private static final DSELVisitor visitor = new DSELVisitor();
+public class DSELConverter {
 
     /**
-     * Utility method to convert a TQL query to a DSEL query.
+     * Utility method to convert a TQL tqlQuery to a DSEL tqlQuery.
      *
-     * @param tqlQuery TQL query as String
+     * @param tqlQuery TQL tqlQuery as String
+     * @param fieldToType a Map object used to get a type (native or semantic type) from a field name, this is a lightweight
+     * representation of the schema
      * @return DSEL ELNode ready to serve for DSEL interpreter
      */
-    public static ELNode convert(final String tqlQuery) throws TqlException {
+    public ELNode convert(final String tqlQuery, Map<String, String> fieldToType) throws TqlException {
         final Expression tqlExpression = Tql.parse(tqlQuery);
-        return convert(tqlExpression);
+        return convert(tqlExpression, fieldToType);
     }
 
     /**
      * Utility method to convert a TQL query to a DSEL query.
      *
      * @param tqlQuery TQL query
+     * @param fieldToType a Map object used to get a type (native or semantic type) from a field name, this is a lightweight
+     * representation of the schema
      * @return DSEL ELNode ready to serve for DSEL interpreter
      */
-    public static ELNode convert(final Expression tqlQuery) throws TqlException {
+    public static ELNode convert(final Expression tqlQuery, Map<String, String> fieldToType) throws TqlException {
```

Comment:
can you add a signature for expressions that do not need the map
```suggestion
    public static ELNode convert(final Expression tqlQuery) throws TqlException
     {
        return convert(tqlQuery, Map.of());
      }
     
```